### PR TITLE
hoedown: new recipe

### PR DESCRIPTION
**Pull request by hoanga**
the following recipe adds support for [hoedown](https://github.com/hoedown/hoedown) which includes the hoedown and smartypants utility as well.

sample run:

```
> uname -a
Haiku shredder 1 hrev55910 Feb 26 2022 07:26:08 x86_64 x86_64 Haiku

> hoedown --version
Built with Hoedown 3.0.7.

> smartypants --version
Built with Hoedown 3.0.7.
```

### CODE REVIEW

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	hoedown$secondaryArchSuffix = $portVersion
+	lib:libhoedown$secondaryArchSuffix = $portVersion
+	cmd:hoedown$secondaryArchSuffix = $portVersion
+	cmd:smartypants$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+SUMMARY_devel="The hoedown development files"
```
Comment (waddlesplash) [inline app-text/hoedown/hoedown-3.0.7.recipe:23]: This isn't needed.

Comment (hoanga) [reply app-text/hoedown/hoedown-3.0.7.recipe]: updated recipe to not include summary for devel (unless this meant something else was unneeded?)

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+REVISION="1"
+SOURCE_URI="https://github.com/hoedown/hoedown/archive/refs/tags/$portVersion.tar.gz"
+CHECKSUM_SHA256="01b6021b1ec329b70687c0d240b12edcaf09c4aa28423ddf344d2bd9056ba920"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	hoedown$secondaryArchSuffix = $portVersion
+	lib:libhoedown$secondaryArchSuffix = $portVersion
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:15]: Provided library is: libhoedown.so.3 so maybe?
```diff
- 	lib:libhoedown$secondaryArchSuffix = $portVersion
+ 	lib:libhoedown$secondaryArchSuffix = ${portVersion%%.*}
```

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	cmd:hoedown$secondaryArchSuffix = $portVersion
+	cmd:smartypants$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	hoedown${secondaryArchSuffix}_devel = $portVersion compat >= 1
+	devel:libhoedown$secondaryArchSuffix = $portVersion compat >= 1
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:25]: No portversion need in line 24, use same version compat for the devel package:
```diff
-	hoedown${secondaryArchSuffix}_devel = $portVersion compat >= 1
-	devel:libhoedown$secondaryArchSuffix = $portVersion compat >= 1
+	hoedown${secondaryArchSuffix}_devel
+	devel:libhoedown$secondaryArchSuffix = ${portVersion%%.*}
```

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	"
+REQUIRES_devel="
+	hoedown$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:34]: No new line here

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+{
+	make
+}
+
+INSTALL()
+{
+	# update install paths
+	sed -i 's,PREFIX = /usr/local',"PREFIX = $prefix", Makefile
+	sed -i 's,BINDIR = $(PREFIX)/include',"BINDIR = $binDir", Makefile
+	sed -i 's,LIBDIR = $(PREFIX)/include',"LIBDIR = $libDir", Makefile
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:50]: These should be `bin` and `lib`, not `include`:
```diff
-	sed -i 's,BINDIR = $(PREFIX)/include',"BINDIR = $binDir", Makefile
-	sed -i 's,LIBDIR = $(PREFIX)/include',"LIBDIR = $libDir", Makefile
+	sed -i 's,BINDIR = $(PREFIX)/bin',"BINDIR = $binDir", Makefile
+	sed -i 's,LIBDIR = $(PREFIX)/lib',"LIBDIR = $libDir", Makefile
```

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	sed -i 's,INCLUDEDIR = $(PREFIX)/include',"INCLUDEDIR = $includeDir", Makefile
+	make DESTDIR=$installDestDir install
+
+	prepareInstalledDevelLib libhoedown
+
+	packageEntries devel \
+		$developDir
+}
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:58]: Add TEST()
```diff
+
+ TEST()
+ {
+ 	make test
+ }

**Review by Begasus — CHANGES_REQUESTED** [review]

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc
+	"
+
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:39]: Somehow this one didn't made the review changes :)
Add requirements for the TEST case:
```diff
+ BUILD_REQUIRES="
+ 	cmd:python3
+ 	cmd:tidy
+ 	"
For python3 you need to edit the `makefile` on line 59, change `python` to `python3` there

File: app-text/hoedown/hoedown-3.0.7.recipe (hunk 2 above)
Comment (Begasus) [reply app-text/hoedown/hoedown-3.0.7.recipe:15]: lib:libhoedown$secondaryArchSuffix = ${portVersion%%.*} should go last in the list, after the cmd's

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+	hoedown$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:37]: Forgot this one:
```diff
- 	cmd:gcc
+ 	cmd:gcc$secondaryArchSuffix
```

Comment (hoanga) [reply app-text/hoedown/hoedown-3.0.7.recipe]: i have made updates that should address the recommendations, thanks!

File: app-text/hoedown/hoedown-3.0.7.recipe [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc$secondaryArchSuffix 
+	"
+
+TEST_REQUIRES="
+	cmd:python3
+	cmd:tidy
+"
```
Comment (Begasus) [inline app-text/hoedown/hoedown-3.0.7.recipe:42]: Need to check this, but looks fine so far, thanks, only one thing at first sight, add a `tab` before `"` at `line 42`

Comment (Begasus) [reply app-text/hoedown/hoedown-3.0.7.recipe:42]: If you push this change things can be merged, check has been fine here (and the TEST) :+1:

Comment (hoanga) [reply app-text/hoedown/hoedown-3.0.7.recipe]: updated with tab

**Comment by Begasus** [comment]
Thanks! :+1:

**Comment by korli** [comment]
For some reason the builders don't pick it up.

**Pull request merged**